bugwarrior/services/bz.py

```diff
@@ -260,9 +276,36 @@ def issues(self):
 
                 extra['needinfo_since'] = pytz.UTC.localize(mod_date)
 
+            if issue['status'] == 'ASSIGNED':
+                extra['assigned_on'] = self._get_assigned_date(issue)
+            else:
+                extra['assigned_on'] = None
+
             issue_obj.update_extra(extra)
             yield issue_obj
 
+    def _get_assigned_date(self, issue):
+        assigned_date = None
+
+        if issue['status'] != 'ASSIGNED':
+            return assigned_date
```

Comment:
This condition will never be met since the method is only called `if issue['status'] == 'ASSIGNED'`.